src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java

```diff
@@ -103,6 +104,7 @@ public BuildingCook(final IColony c, final BlockPos l)
         keepX.put(ItemStackUtils.ISFOOD, new Tuple<>(STACKSIZE, true));
         keepX.put(ItemStackUtils.ISCOOKABLE, new Tuple<>(STACKSIZE, true));
         keepX.put(stack -> isAllowedFuel(stack), new Tuple<>(STACKSIZE, true));
+        keepX.put(stack -> !ItemStackUtils.isEmpty(stack.getContainerItem()), new Tuple<>(STACKSIZE, false));
     }
```

Comment:
Will the cook keep all container items for forever? 20 buckets of milk or water too?

Reply:
I should probably exclude milk and water. Probably by excluding buckets. I want to replace this with the custom data load you and I talked about eventually, but realized that this was pretty close to what was needed for now.